### PR TITLE
Rename `oak::grpc::OakNode` to `oak::grpc::ServerNode` 

### PR DESCRIPTION
**Pull request by rbehjati**
# Checklist

- [x] Pull request affects core Oak functionality (e.g. runtime, SDK, ABI)
  - [ ] I have written tests that cover the code changes.
  - [ ] I have checked that these tests are run by [Cloudbuild](cloudbuild.yaml)
  - [x] I have updated [documentation](docs/) accordingly.
  - [ ] I have raised an [issue](https://github.com/project-oak/oak/issues) to
        cover any TODOs and/or unfinished work.
- [ ] Pull request includes prototype/experimental work that is under
      construction.

# Notes
- A few additional files were updated after I ran the `format` script, and they are included in this PR. 
- The rust documentation needs to be regenerated after this PR is merged. I did not create a new issue for it. Do we need one?

### CODE REVIEW

**Comment by tiziano88** [comment]
FYI I think this conflicts with #628 , which was already merged earlier today, so you would need to rebase on top of that (or rather, on top on master, usually).

**Comment by tiziano88** [comment]
> A few additional files were updated after I ran the format script, and they are included in this PR.

I'm not sure how that could have happened. Maybe you ran the format script from outside Docker?

> The rust documentation needs to be regenerated after this PR is merged. I did not create a new issue for it. Do we need one?

No need for the issue, just run https://github.com/project-oak/oak/blob/master/scripts/build_gh_pages after this PR is merged.

**Comment by tiziano88** [comment]
Could you use rebase instead of merge so that it's clearer what commits belong to this PR, and merging is hopefully more predictable? Thanks

File: sdk/rust/oak/src/grpc/mod.rs [hunk 1]
```diff
@@ -119,14 +119,14 @@ impl ChannelResponseWriter {
 
 /// Trait for Oak Nodes that act as a gRPC services.
 ///
-/// An `OakNode` instance is normally passed to [`oak::run_event_loop`], to
+/// An `ServerNode` instance is normally passed to [`oak::run_event_loop`], to
```
Comment (tiziano88) [inline sdk/rust/oak/src/grpc/mod.rs]:
```suggestion
/// A `ServerNode` instance is normally passed to [`oak::run_event_loop`], to
```

**Comment by rbehjati** [comment]
> thanks! Between the PR message and the various commit, I am not sure what the final commit message you are intending to use is. Perhaps squash your own commits together and give the final commit the message you would like to use for merging? Unless there is a nicer way of doing it that I'm not aware of :)

Sure! My intention was to squash when merging. I can squash first :)

**Pull request merged**